src/database/sqlite/schema.rs

```diff
@@ -57,7 +59,8 @@ CREATE TABLE coins (
  * we can get the derivation index from the parent descriptor from bitcoind.
  */
 CREATE TABLE addresses (
-    address TEXT NOT NULL UNIQUE,
+    receive_address TEXT NOT NULL UNIQUE,
+    change_address TEXT NOT NULL UNIQUE,
```

Comment:
why having row in database with the tuple (receive and change) and not having a `is_change` bool ?

Reply:
Maybe i misunderstand your question, but you need an address->derivation index mapping. You necessarily need a key per address in this mapping.

Reply:
I meant why storing:
| receive_address | change_address | derivation_index |
|------------------|------------------------|--------------------------|
| addr1 | addr2 | 1|

and not
 | address | is_change | derivation_index |
|------------------|------------------------|--------------------------|
| addr1 | false | 1|
| addr2 | true | 1|

Reply:
Because it's wasteful? That's 2 added integer per derivation index. What advantage would it have?

Reply:
To only store address that we use, but maybe that is not our process:
If we do no spend at all and receive only deposit until derivation_index = 10, does it means that we want to store the 10 change addresses that are not used at all ?

Reply:
This mapping is for detecting coins up to the gap limit, not to store addresses we used. Necessarily we need to have addresses up to the gap limit.

>  If we do no spend at all and receive only deposit until derivation_index = 10, does it means that we want to store the 10 change addresses that are not used at all ?

Are we sure we are *never* ever going to receive a payment on an address before derivation index 10? I don't think so at all. What if we gave away this address already? We trust others to not use it? What if we didn't but another wallet did? What if we recovered from backup? Etc..